test-storybooks/portable-stories-kitchen-sink/svelte/stories/__snapshots__/Button.test.ts.snap

```diff
@@ -12,17 +12,14 @@ exports[`Renders CSF2Secondary story 1`] = `
        
     </button>
     
-    
   </div>
 </body>
 `;
 
 exports[`Renders CSF2StoryWithParamsAndDecorator story 1`] = `
 <body>
   <div>
-    <div
-      style="margin: 3em;"
-    >
+    <decorator.component>
```

Comment:
This exposes a bug in Testing Library Svelte and/or Svelte 4.

When you use `<something.theComponent />`, it will snapshot as that directly instead of what the component actually is, as we see in the diff here. I confirmed that it is unrelated to Storybook/Portable Stories and that it's only a problem in Svelte 4. I wrote this simple test setup:
```svelte
// Wrapper.svelte
<div style="margin: 3em;">
  <slot />
</div>
```

```svelte
// Inner.svelte
<script>
  export let stuff;
</script>

<stuff.Component> hello </stuff.Component>
```

```js
// the.test.js
import { it, expect } from 'vitest';
import { render } from '@testing-library/svelte';
import Inner from './Inner.svelte';
import Wrapper from './Wrapper.svelte';

it('barebone', () => {
  render(Inner, { stuff: { Component: Wrapper }});
  expect(document.body).toMatchInlineSnapshot();
});
```

Using `<stuff.Component> hello </stuff.Component>` it had this weird flaw in the snapshot, using `<svelte:component this={stuff.Component}>hello</svelte:component>` it didn't.

I think this is okay honestly, not much we can do about it other than not fixing the warning here. Thoughts @yannbf @kasperpeulen?

Reply:
It was discussed with @JReinhold that he will try to apply version-dependent logic so that the decorator.Component is only used in Svelte 5